mypy/newsemanal/semanal.py

```diff
@@ -2390,21 +2397,28 @@ def check_and_set_up_type_alias(self, s: AssignmentStmt) -> bool:
         alias_node = TypeAlias(res, self.qualified_name(lvalue.name), s.line, s.column,
                                alias_tvars=alias_tvars, no_args=no_args)
         if existing:
-            # Did alias get updated?
-            if (isinstance(existing.node, PlaceholderNode) or
-                    isinstance(existing.node, TypeAlias) and existing.node.target != res):
-                self.progress = True
-                # We need to defer so that this change can get propagated to base classes.
-                self.defer()
+            # An alias gets updated.
+            if self.final_iteration:
+                # Error will be reported elsewhere, but we can't make progress during
+                # final iteration.
+                return True
```

Comment:
This part (check for final iteration) looks a bit suspicious. Could you please add a bit longer comment about what kind of situation may cause this, and what error will be reported (will this be just invalid type in assignment, I noticed a TODO about this)?

Reply:
Added an explicit error message, since it wasn't clear if an error will be reported elsewhere.